components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java

```diff
@@ -157,87 +157,42 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
             throws APIMgtUsageQueryServiceClientException {
 
         List<String> subscriberApps = getAppsbySubscriber(subscriberName, groupId);
-        String[] stringArray = subscriberApps.toArray(new String[subscriberApps.size()]);
+        String concatenatedKeySetString = "";
 
-        String concatenatedKeySetString = "''";
-
-        if (stringArray.length > 0) {
-            concatenatedKeySetString = buildKeySetString(stringArray);
+        int size = subscriberApps.size();
+        if (size > 0) {
+            concatenatedKeySetString += "'" + subscriberApps.get(0) + "'";
+        } else {
+            return new ArrayList<APIResponseFaultCountDTO>();
```

Comment:
Would be much elegant to use "Collections.emptyList()" instead of instantiating a new array list as done above.